ml-agents/mlagents/trainers/tests/test_simple_rl.py

```diff
@@ -262,28 +182,45 @@ def _check_environment_trains(
 
         # Begin training
         tc.start_learning(env_manager)
-        print(tc._get_measure_vals())
         if (
             success_threshold is not None
         ):  # For tests where we are just checking setup and not reward
-            for mean_reward in tc._get_measure_vals().values():
-                assert not math.isnan(mean_reward)
-                assert mean_reward > success_threshold
+
+            processed_rewards = [
+                reward_processor(rewards) for rewards in env.final_rewards.values()
+            ]
+            assert all(not math.isnan(reward) for reward in processed_rewards)
+            if not check_fail:
+                assert all(reward > success_threshold for reward in processed_rewards)
+            else:
+                assert any(
+                    reward > success_threshold for reward in processed_rewards
```

Comment:
If this is only going to be used for the ghost training test, maybe it would be better to leave out of the general function and just grab the rewards from `env` in `test_simple_ghost_fails()`

Reply:
True. This is actually only special casing for the ghost failure test which I agree should be done elsewhere.  I can remove the check_fail flag too.